stingray/events.py

```diff
@@ -0,0 +1,212 @@
+"""
+Definition of :class:`EventList`.
+
+:class:`EventList` is used to handle photon arrival times.
+"""
+from __future__ import (absolute_import, unicode_literals, division,
+                        print_function)
+
+from stingray.simulator.events import events_to_lc, lc_to_events, assign_energies
+
+from .lightcurve import Lightcurve
+import stingray.io as io
+import stingray.utils as utils
+
+import numpy as np
+
+
+class EventList(object):
+    def __init__(self, times, energies=None, mjdref=0, dt=0, notes="", gti=None, pi=None,
+                 pha=None):
+        """
+        Make an event list object from an array of time stamps
+
+        Parameters
+        ----------
+        times: iterable
+            A list or array of time stamps
+
+        Other Parameters
+        ----------------
+        dt: float
+            The time resolution of the events. Only relevant when using events
+            to produce light curves with similar bin time.
+
+        energies: iterable
+            A list of array of photon energy values
+
+        mjdref : float
+            The MJD used as a reference for the time array.
+
+        gtis: [[gti0_0, gti0_1], [gti1_0, gti1_1], ...]
+            Good Time Intervals
+
+        pi : integer, numpy.ndarray
+            PI channels
+
+        pha : float, numpy.ndarray
+            Photon energies
+
+        Attributes
+        ----------
+        time: numpy.ndarray
+            The array of event arrival times, in seconds from the reference
+            MJD (self.mjdref)
+
+        energies: numpy.ndarray
+            The array of photon energy values
+
+        ncounts: int
+            The number of data points in the event list
+
+        dt: float
+            The time resolution of the events. Only relevant when using events
+            to produce light curves with similar bin time.
+
+        mjdref : float
+            The MJD used as a reference for the time array.
+
+        gtis: [[gti0_0, gti0_1], [gti1_0, gti1_1], ...]
+            Good Time Intervals
+
+        pi : integer, numpy.ndarray
+            PI channels
+
+        pha : float, numpy.ndarray
+            Photon energies
+
+        """
+        self.time = np.array(times, dtype=np.longdouble)
+        self.energy = np.array(energies)
+        self.ncounts = len(times)
+        self.notes = notes
+        self.dt = dt
+        self.mjdref = mjdref
+        self.gti = gti
+        self.pi = pi
+        self.pha = pha
+
+    @staticmethod
+    def from_fits(fname, **kwargs):
+        from .utils import _assign_value_if_none
```

Comment:
I picked `from_fits()` method from @matteobachetti's commit. However, I think it would be more appropriate to have it as a part of `io.py`. I will set up a new PR for that.